packages/manager/apps/pci-private-network/src/pages/new/private-network/PrivateNetworkConfig.component.tsx

```diff
@@ -0,0 +1,252 @@
+
+  const onDefineVlanId = (event: CustomEvent) => {
+    const value = event.detail.checked;
+
+    if (!value) {
+      unregister('vlanId'); // customer doesn't want to define it then reset to default
+    } else {
+      setValue('vlanId', defaultVlanId);
+    }
+
+    setDefineVlanId(value);
+  };
+
+  const onChangeVlanId = ({ target }) => {
+    const id = target.value as number;
+    const cidr = getDefaultCIDR(id);
+    setValue('vlanId', id, {
+      shouldValidate: true,
+    });
+    setValue('subnet.cidr', cidr);
+  };
+
+  return (
+    <div className="flex flex-col gap-6 my-8">
+      <Subtitle>
+        {t('pci_projects_project_network_private_create_configure')}
+      </Subtitle>
+      <OsdsFormField
+        data-testid="private-network-name-error"
+        error={hasError ? t('common:common_field_error_required') : ''}
+      >
+        <OsdsText color={ODS_TEXT_COLOR_INTENT.text} slot="label">
+          {t('pci_projects_project_network_private_create_name')}
+        </OsdsText>
+
+        <OsdsInput
+          className="md:w-2/5"
+          name="privateNetworkName"
+          data-testid="private-network-name"
+          type={ODS_INPUT_TYPE.text}
+          color={
+            hasError
+              ? ODS_THEME_COLOR_INTENT.error
+              : ODS_THEME_COLOR_INTENT.primary
+          }
+          onOdsInputBlur={() => {
+            if (!name) {
+              setValue('name', '', {
+                shouldValidate: true,
+                shouldTouch: true,
+              });
+            }
+          }}
+          onOdsValueChange={({ target }) =>
+            setValue('name', target.value as string, {
+              shouldValidate: true,
+              shouldTouch: true,
+            })
+          }
+          error={hasError}
+        />
+      </OsdsFormField>
+
+      {!onLocalZone && (
+        <div className="flex flex-col gap-5">
+          <OsdsText
+            level={ODS_TEXT_LEVEL.subheading}
+            color={ODS_TEXT_COLOR_INTENT.primary}
+            hue={ODS_TEXT_COLOR_HUE._800}
+          >
+            {t('pci_projects_project_network_private_create_layer_2_options')}
+          </OsdsText>
+          <OsdsCheckbox
+            data-testid="define-vlan"
+            onOdsCheckedChange={onDefineVlanId}
+          >
+            <OsdsCheckboxButton
+              size={ODS_CHECKBOX_BUTTON_SIZE.sm}
+              color={ODS_THEME_COLOR_INTENT.primary}
+            >
+              <span slot="end">
+                <OsdsText
+                  color={ODS_TEXT_COLOR_INTENT.text}
+                  level={ODS_TEXT_LEVEL.body}
+                  size={ODS_TEXT_SIZE._400}
+                >
+                  {t(
+                    'pci_projects_project_network_private_create_configure_choose_vlan',
+                  )}
+                </OsdsText>
+              </span>
+            </OsdsCheckboxButton>
+          </OsdsCheckbox>
+
+          {defineVlanId && (
```

Comment:
As an improvement, JSX could be split into smaller components to ensure readability and testability (non-blocking)